aptos-move/framework/aptos-stdlib/sources/cryptography/algebra.move

```diff
@@ -0,0 +1,70 @@
+module aptos_std::algebra {
+    use std::option::Option;
+
+    struct BLS12_381_Fr has copy, drop { handle: u64 }
+    struct BLS12_381_Fq has copy, drop { handle: u64 }
+    struct BLS12_381_Fq2 has copy, drop { handle: u64 }
+    struct BLS12_381_Fq6 has copy, drop { handle: u64 }
+    struct BLS12_381_Fq12 has copy, drop { handle: u64 }
+    struct BLS12_381_G1 has copy, drop { handle: u64 }
+    struct BLS12_381_G2 has copy, drop { handle: u64 }
+    struct BLS12_381_Gt has copy, drop { handle: u64 }
+
```

Comment:
all the functions below, they don't need to be native. I'm just saving some lines here since this is a preview...